### DISCUSSION TITLE
Question: what do the numbers in the file name represent?

### DISCUSSION THREAD

**Discussion by rdong8** [hanabi1224/Programming-Language-Benchmarks#255] [Q&A]
Like does 1.rs and 1.cpp mean that the two programs use the same algorithm?

**Comment by rschwietzke** [comment]
I also stumbled upon that. Shouldn't we start to name the test case something like Mandelbrot-01.java so we can see the test algorithm quickly, match naming conventions of the language (class name and similar, not relevant for all languages) and being able to create different types of impl to show the influence of language concepts?

**Comment by DRigby26** [comment]
I am curious about this as well.

**Reply by cyrusmsk** [reply]
I think the names comes from original project: https://benchmarksgame-team.pages.debian.net/benchmarksgame/performance/mandelbrot.html

there it uses different optimisations (like parallelism, simd, or different approach).
In hanabi’s project the idea was to use the same algo and approach, but sometimes there are several ways to implement even the same approach (use bitarray or array of booleans for example).
For parallel version - use -m flag
For SIMD version - use -i flag

there is still room for improvements to bring all versions of the codes to the identical approaches. So if you will find something - just write about it or even better make a PR
